### PR TITLE
BOAC-539, demo-mode-blur for names in matrix view; student names in search replaced with random gen

### PR DESCRIPTION
**Pull request by johncrossman**
https://jira.ets.berkeley.edu/jira/browse/BOAC-539

One more bit of blurring.

### CODE REVIEW

File: boac/static/app/shared/utilService.js [hunk 1]
```diff
@@ -78,11 +78,20 @@
       return str.replace(/([a-z])([A-Z])/g, '$1-$2').toLowerCase();
     };
 
+    /**
+     * @param  {String}     str      Word, phrase or whatever
+     * @return {String}              Unrecognizable representation of str
+     */
+    var obfuscate = function(str) {
+      return btoa(str);
```
Comment (raydavis) [inline boac/static/app/shared/utilService.js]: Using "btoa" as an obfuscation technique worries me a little. Deobfuscation is trivial, but it's easy to slip into a false sense of security.

Comment (paulkerschen) [reply boac/static/app/shared/utilService.js]: Assuming obfuscation doesn't need to be repeatable, perhaps random strings are best? Everyone's got their own favorite way to do that client-side, looks like: https://stackoverflow.com/questions/1349404/generate-random-string-characters-in-javascript

**Comment by johncrossman** [comment]
A proper `demo-mode` impl might use something like http://chancejs.com or server/db side obfuscation. This impl was hurried and, for now, might qualify as "good enough".

**Comment by raydavis** [comment]
Thanks, @johncrossman .

**Pull request merged**